[#12281] Fix deleted feedback sessions table overflow

Sho-Shoo opened the pull request:
<!-- Before opening a PR, please ensure you have read our contributor guidelines -->
<!-- at https://teammates.github.io/teammates/process.html#step-4-submit-a-pr. -->

<!-- PR title: Copy-and-paste the name of the issue this PR is fixing, -->
<!-- and include the issue number in front in square brackets. -->
<!-- e.g. [#3942] Remove unnecessary System.out.printlns from Java files -->

<!-- Add the issue number to the "Fixes" keyword below. -->
Fixes #12281 

**Outline of Solution**

<!-- Give a brief description of how you solved the issue. -->
<!-- If the solution includes any changes in UI, do also attach screenshots of the new UI. --> 

- Fixed the overflow issue of action cell by adding `overflow-auto` class tag. 
- Fixed background color issue by adding `d-flex` class tag. 

Below is updated UI. 

<img width="494" alt="fied-ui" src="https://user-images.githubusercontent.com/41343828/233820096-51bf8f3b-f4ed-4516-a5f4-a42f008aa0a0.png">

Sho-Shoo commented:
Hi! Simply wondering if it is alright to fail the E2E testing? Since a lot recent commits on the main branch I see are also failing the same check.

zhaojj2209 commented:
@Sho-Shoo The TEAMMATES E2E test workflow occasionally need reruns, maintainers will help to monitor the tests and rerun them if necessary.

While this PR solves the overflow and background colour issue, as can be seen in the screenshot, some of the columns are rather unreadable. Shall we set a minimum width for the table so that each table column can be read more easily? You may reference similar approaches for other tables in TEAMMATES (e.g. notifications table).

Sho-Shoo commented:
Fixes #12281 

Outline of Solution

- Fixed the overflow issue of action cell by adding `overflow-auto` class tag.
- Fixed background color issue by adding `d-flex` class tag.
- Added minimum width of `deleted-sessions-table` to enable horizontal scrolling. 

Below is updated UI.


![demo2](https://user-images.githubusercontent.com/41343828/235373197-48a6e4fc-eb5a-4834-8b97-33f9f4c27a6f.gif)

jasonqiu212 commented on src/web/app/components/sessions-recycle-bin-table/sessions-recycle-bin-table.component.scss:
```diff
@@ -47,3 +47,7 @@
 }
+
+#deleted-sessions-table {
+  min-width: 700px;
```
Maybe we can move this attribute to the `.recycle-bin-table` class in line 5? so that we can reuse the CSS classes

jasonqiu212 commented on src/web/app/components/sessions-recycle-bin-table/sessions-recycle-bin-table.component.html:
```diff
@@ -75,14 +75,16 @@ <h2 class="text-muted">
+            <td class="text-center actions-cell">
+              <div class="d-flex">
+                <button id="btn-restore-{{ i }}" type="button" class="btn btn-light btn-sm" (click)="restoreSessionEvent.emit(recycleBinFeedbackSessionRowModel)">Restore</button>
+                <button id="btn-delete-{{ i }}" type="button" class="btn btn-light btn-sm text-danger align-row" (click)="permanentlyDeleteSessionEvent.emit(recycleBinFeedbackSessionRowModel); setRowClicked(i)">
```
I think we can remove the `align-row` class here, since it adds a redundant padding inside the button and causes the text to not be centered.

<img width="719" alt="Screen Shot 2023-05-02 at 8 47 30 AM" src="https://user-images.githubusercontent.com/25262042/235556272-b5a486d7-9490-4664-87e8-252589999630.png">

jasonqiu212 replied:
By doing so, we can also delete the `align-row` CSS class in the SCSS file

Sho-Shoo replied:
Thank you for letting me know. I have made the changes according to your review. The updated UI does not have the margin now: 
![截屏2023-05-01 下午9 19 31](https://user-images.githubusercontent.com/41343828/235559501-33c976f8-8d4a-4169-a0ab-47da188bf9e6.png)

jasonqiu212 reviewed (changes requested):
Hi @Sho-Shoo, Thank you for your contribution! Your solution works well to address the issue, just some nits here and there to avoid redundancy :)

Sho-Shoo commented:
Hi! I have made all changes requested by the reviewer. Please checkout the most recent push.

jasonqiu212 commented on src/web/app/components/sessions-recycle-bin-table/sessions-recycle-bin-table.component.scss:
```diff
@@ -4,6 +4,7 @@
 
```
As mentioned, we can now remove the `.align-row` CSS class in line 42, since it's not used anywhere

Sho-Shoo replied:
Sorry for omitting that in the previous commit. The recent commit should resolve this.

jasonqiu212 reviewed (approved):
LGTM! thanks for the changes
